### DISCUSSION TITLE
[@types/passport-apple] Invalid third argument of VerifyFunction

### DISCUSSION THREAD

**Discussion by wojtekmaj** [DefinitelyTyped/DefinitelyTyped#66925] [Issues with a @types package]
VerifyFunction's third argument is supposed to be DecodedIdToken. In passport-apple 2.0.2, it's not: it's an encoded JWT token.

Exact commit when that happened: https://github.com/ananay/passport-apple/commit/6b8c1ce0fecf6cac2c8a0f669824dc2640fb5112

[a comment by an automated account is not shown]

**Comment by Semigradsky** [comment]
@wojtekmaj I can send PR, but I am not sure about type `VerifyCallback`:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e711b460e132f737f1080eb3f4053b887870f11d/types/passport-apple/index.d.ts#L45
It still require `user?: object, info?: object` params?

In example I see `cb(null, idToken);`, looks like only one parameter with string type.

**Reply by wojtekmaj** [reply]
I think in general you would like to return an object to Passport.

I don't see anything that would make me think it would require updating:

https://github.com/ananay/passport-apple/compare/v1.1.1...2.0.2

Also, in real life application, that's the only issue I've stumbled upon. The rest, when I do what TypeScript says, works just fine as is.

**Reply by Semigradsky** [reply]
PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/66929
